### ISSUE TITLE
tiny-parser.cc: -Wformat-diag messages

### ISSUE THREAD

**Issue by HSorensen** [GCC-Tiny/gcc-tiny-docs-rtd#43]
Steps to reproduce:

```shell
cd gcc-build
touch ../gcc/gcc/tiny/tiny-parser.cc
make
```

gives output

```
../../gcc/gcc/tiny/tiny-parser.cc: In member function ‘Tiny::const_TokenPtr Tiny::Parser::expect_token(Tiny::TokenId)’:
../../gcc/gcc/tiny/tiny-parser.cc:218:60: warning: unquoted whitespace character ‘\x0a’ in format [-Wformat-diag]
  218 |       error_at (t->get_locus (), "expecting %s but %s found\n",
      |                                                            ^~
../../gcc/gcc/tiny/tiny-parser.cc: In member function ‘void Tiny::Parser::unexpected_token(Tiny::const_TokenPtr)’:
../../gcc/gcc/tiny/tiny-parser.cc:233:46: warning: unquoted whitespace character ‘\x0a’ in format [-Wformat-diag]
  233 |   ::error_at (t->get_locus (), "unexpected %s\n", t->get_token_description ());
      |                                              ^~
../../gcc/gcc/tiny/tiny-parser.cc: In member function ‘Tiny::Tree Tiny::Parser::null_denotation(Tiny::const_TokenPtr)’:
../../gcc/gcc/tiny/tiny-parser.cc:1556:51: warning: unbalanced punctuation character ‘)’ in format [-Wformat-diag]
 1556 |           error_at (tok->get_locus (), "expecting ) but %s found\n",
      |                                                   ^
../../gcc/gcc/tiny/tiny-parser.cc:1556:65: warning: unquoted whitespace character ‘\x0a’ in format [-Wformat-diag]
 1556 |           error_at (tok->get_locus (), "expecting ) but %s found\n",
      |                                                                 ^~
../../gcc/gcc/tiny/tiny-parser.cc:1571:54: warning: unquoted keyword ‘int’ in format [-Wformat-diag]
 1571 |                       "operand of unary plus must be int or float but it is %s",
      |                                                      ^~~
../../gcc/gcc/tiny/tiny-parser.cc:1571:61: warning: unquoted keyword ‘float’ in format [-Wformat-diag]
 1571 |                       "operand of unary plus must be int or float but it is %s",
      |                                                             ^~~~~
../../gcc/gcc/tiny/tiny-parser.cc:1588:47: warning: unquoted keyword ‘int’ in format [-Wformat-diag]
 1588 |               "operand of unary minus must be int or float but it is %s",
      |                                               ^~~
../../gcc/gcc/tiny/tiny-parser.cc:1588:54: warning: unquoted keyword ‘float’ in format [-Wformat-diag]
 1588 |               "operand of unary minus must be int or float but it is %s",
      |                                                      ^~~~~
../../gcc/gcc/tiny/tiny-parser.cc: In member function ‘bool Tiny::Parser::check_logical_operands(Tiny::const_TokenPtr, Tiny::Tree, Tiny::Tree)’:
../../gcc/gcc/tiny/tiny-parser.cc:1866:72: warning: unquoted whitespace character ‘\x0a’ in format [-Wformat-diag]
 1866 |         "operands of operator %s must be boolean but they are %s and %s\n",
      |                                                                        ^~
```

This code and blog need to be updated to remove this warning.

- unquoted whitespace character ‘\x0a’
  - error messages should not contain newlines
-  unquoted keyword ‘float’ and ´int´
  - Add proper quotes
- unbalanced punctuation character ‘)’

**Comment by HSorensen** [comment]
These must be some new warnings. I think the fixes are straight forward. Just need to ensure the blog part04 where the code for the tiny_parse.cc is discussed.

After first pass of clean up:

```
../../gcc/gcc/tiny/tiny-parser.cc: In member function ‘Tiny::Tree Tiny::Parser::null_denotation(Tiny::const_TokenPtr)’:
../../gcc/gcc/tiny/tiny-parser.cc:1556:51: warning: unbalanced punctuation character ‘)’ in format [-Wformat-diag]
 1556 |           error_at (tok->get_locus (), "expecting ) but %s found",
      |                                                   ^
../../gcc/gcc/tiny/tiny-parser.cc:1571:54: warning: unquoted keyword ‘int’ in format [-Wformat-diag]
 1571 |                       "operand of unary plus must be int or float but it is %s",
      |                                                      ^~~
../../gcc/gcc/tiny/tiny-parser.cc:1571:61: warning: unquoted keyword ‘float’ in format [-Wformat-diag]
 1571 |                       "operand of unary plus must be int or float but it is %s",
      |                                                             ^~~~~
../../gcc/gcc/tiny/tiny-parser.cc:1588:47: warning: unquoted keyword ‘int’ in format [-Wformat-diag]
 1588 |               "operand of unary minus must be int or float but it is %s",
      |                                               ^~~
../../gcc/gcc/tiny/tiny-parser.cc:1588:54: warning: unquoted keyword ‘float’ in format [-Wformat-diag]
 1588 |               "operand of unary minus must be int or float but it is %s",
      |                                                      ^~~~~
```

**Comment by HSorensen** [comment]
Google found an old patch: https://marc.info/?l=gcc-patches&m=111421466916257&w=2

Looks like the "quoted" can be achieved by using %< ... %>

So

```
error_at (
         tok->get_locus (),
         "operand of unary minus must be %<int%> or %<float%> but it is '%s'",
         print_type (expr.get_type ()));

```

**Comment by SamuraiCrow** [comment]
Can be closed after PR #44 if review is ok.  I'm not so familiar with the indention style.  Should parameters be outdented to the curly brace level after the function call itself is indented farther?
